exporter/management/commands/flattener.py

```diff
@@ -27,42 +27,52 @@ class Command(BaseCommand):
     """
 
     def handle(self, *args, **options):
-        consume(callback, "flattener_init", decorator=decorator)
+        consume(callback, "flattener_init", ["flattener_init", "flattener_file"], decorator=decorator)
 
 
 def callback(state, channel, method, properties, input_message):
     job_id = input_message.get("job_id")
-    max_rows_lower_bound = Job.objects.get(id=job_id).get_max_rows_lower_bound()
-
+    file_path = input_message.get("file_path")
     export = Export(job_id, export_type="flat")
-    export.lock()
 
     # Acknowledge now to avoid connection losses. The rest can run for hours and is irreversible anyhow.
     ack(state, channel, method.delivery_tag)
 
-    for entry in os.scandir(export.directory):
-        if not entry.name.endswith(".jsonl.gz") or "_" in entry.name:  # don't process months at the moment
-            continue
+    if file_path:
+        process_file(file_path, job_id)
+    else:
+        for entry in os.scandir(export.directory):
+            if not entry.name.endswith(".jsonl.gz") or "_" in entry.name:  # don't process months at the moment
+                continue
+            publish({"job_id": job_id, "file_path": entry.path}, "flattener_file")
+
+
+def process_file(file_path, job_id):
+    max_rows_lower_bound = Job.objects.get(id=job_id).get_max_rows_lower_bound()
+    file_name = Path(file_path).name
+
+    export = Export(job_id, export_type="flat", lockfile_suffix=file_name)
+    export.lock()
 
-        with tempfile.TemporaryDirectory() as tmpdirname:
-            tmpdir = Path(tmpdirname)
-            infile = tmpdir / entry.name[:-3]  # remove .gz
-            outdir = tmpdir / "flatten"  # force=True deletes this directory
-            final_path_prefix = entry.path[:-9]  # remove .jsonl.gz
+    with tempfile.TemporaryDirectory() as tmpdirname:
```

Comment:
Since we've witnessed errors raised by tar, we can put a try/except/finally around this whole block, to ensure unlock always runs even in case of an error.